### PR TITLE
Update documentation for pybids

### PR DESCRIPTION
**Pull request by kaitj**
## Proposed changes

Adds some documentation for pybids.

- Add docstring to _gen_bids_layout
- Update bids_app/config.rst doc for new pybids variables
- Add instructions to "Running snakebids"

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you are unsure about any of the choices, don't hesitate to ask!

- [ ] Changes have been tested to ensure that fix is effective or that a feature works.
- [ ] Changes pass the unit tests
- [ ] I have included necessary documentation or comments (as necessary)
- [ ] Any dependent changes have been merged and published

## Notes
All PRs will undergo the unit testing before being reviewed. You may be requested to explain or make additional changes before the PR is accepted.

### CODE REVIEW

File: snakebids/core/input_generation.py [hunk 1]
```diff
@@ -473,6 +473,29 @@ def _gen_bids_layout(
...
+    ----------
+    bids_dir : str
+        Path to bids directory
+
+    derivatives : bool
+        A boolean (or path(s) to derivatives datasets) that
+        determines whether snakebids will search in the
+        derivatives subdirectory of the input dataset.
+
+    pybids_database_dir : str
```
Comment (pvandyken) [inline snakebids/core/input_generation.py:487]: These should actually go under the public `generate_inputs()` function. Either copy/paste, or just move

Comment (kaitj) [reply snakebids/core/input_generation.py:495]: Copied over in 17a47f8

**Review by pvandyken — APPROVED** [review]

**Review by tkkuehn — CHANGES_REQUESTED** [review]
One minor comment otherwise looks good to me!

File: docs/running_snakebids/overview.md [hunk 2]
```diff
@@ -7,11 +7,16 @@ Snakebids apps generated with the cookiecutter template will have a simple execu
...
+Note that if any rules in the Snakebids workflow use Singularity containers, special precautions must be taken to ensure the input dataset is bound to the Singularity environment. Either:
 
 1. Inputs are copied into a working subdirectory of the output directory before any processing that requires a Singularity container is performed, or:
 2. The `SINGULARITY_BINDPATH` environment variable binds the location of the input dataset.
 
+Indexing of large datasets can be a time-consuming process. Snakebids, through `PyBIDS` has the ability to create or leverage an existing database, requiring indexing of datasets to be only performed when user chooses to do so (usually if the dataset has changed)! Note, this feature is **opt-in**, meaning it is not used unless the associated config variables are used. To opt-in:
+
+1. Uncomment the lines in `snakebids.yml` containing `pybids_db_dir` and `pybids_db_reset`.
+1. The variables can be updated directly in this file or through the CLI by using `--config pybids_db_dir={path}` to update the database path and `--config pybids_db_reset={True/False}` to indicate whether the database should be updated.
```
[the remark this answers is not in the document]
Comment (kaitj) [reply docs/running_snakebids/overview.md]: Fixed in df1fc94.

File: docs/running_snakebids/overview.md [hunk 3]
```diff
@@ -7,11 +7,16 @@ Snakebids apps generated with the cookiecutter template will have a simple execu
...
+Note that if any rules in the Snakebids workflow use Singularity containers, special precautions must be taken to ensure the input dataset is bound to the Singularity environment. Either:
 
 1. Inputs are copied into a working subdirectory of the output directory before any processing that requires a Singularity container is performed, or:
 2. The `SINGULARITY_BINDPATH` environment variable binds the location of the input dataset.
 
+Indexing of large datasets can be a time-consuming process. Snakebids, through `PyBIDS` has the ability to create or leverage an existing database, requiring indexing of datasets to be only performed when user chooses to do so (usually if the dataset has changed)! Note, this feature is **opt-in**, meaning it is not used unless the associated config variables are used. To opt-in:
+
+1. Uncomment the lines in `snakebids.yml` containing `pybids_db_dir` and `pybids_db_reset`.
+1. The variables can be updated directly in this file or through the CLI by using `-pybidsdb-dir {dir}` to update the database path and `--reset-db` to indicate whether the database should be updated. _Note: CLI arguments take precendence if both CLI and config variables are set._
```
Comment (tkkuehn) [inline docs/running_snakebids/overview.md:18]:
```suggestion
1. The variables can be updated directly in this file or through the CLI by using `-pybidsdb-dir {dir}` to specify the database path and `--reset-db` to indicate that the database should be updated. _Note: CLI arguments take precendence if both CLI and config variables are set._
```
Nitpick to clarify that there doesn't need to be an existing database path to update and that `--reset-db` is a store_true argument.

**Review by tkkuehn — CHANGES_REQUESTED** [review]

File: docs/running_snakebids/overview.md (hunk 3 above)
Comment (kaitj) [reply docs/running_snakebids/overview.md]: Committed suggestions in 7336a1a

**Review by tkkuehn — APPROVED** [review]

**Pull request merged**